### DISCUSSION TITLE
working scrobbler clients

### DISCUSSION THREAD

**Discussion by 1xPdd** [krateng/maloja#89] [General]
Hey, great project. Thanks.

I saw in the Readme that you wanted info about working clients.

So far, I have:

- Simple Scrobbler (Android). Works using the listenbrainz API
- DeaDBeeF (Linux). No success. I tried all the APIs as well as any permutation I could think of. Not sure what logs would show errors that would help finding the exact problem.

**Comment by FoxxMD** [comment]
I'll add that [airsonic-advanced](https://github.com/airsonic-advanced/airsonic-advanced) works as well using the listenbrainz protocol:

- Open settings -> credentials
- Add credentials with app Listenbrainz
- Password is any maloja token
![image png e6c35ca339726145d3a6d94e09949fe3](https://user-images.githubusercontent.com/4663766/96608072-f947f800-12c6-11eb-8f50-40e2ec94904f.png)

- Open settings -> personal
- Check "Register what I'm playing at ListenBrainz"
- In the ListenBrainz URL field that appears enter the same URL as used for web-scrobbler
![image png e3bb702d8c58e1659696df6006ae31b0](https://user-images.githubusercontent.com/4663766/96608109-01a03300-12c7-11eb-947e-a82f5b9db1f9.png)

**Comment by krateng** [comment]
Thank you! Is Deadbeef using the legacy protocol by any chance?

**Comment by 1xPdd** [comment]
> Is Deadbeef using the legacy protocol by any chance?

I haven't a clue, sorry. This seems to be the source of the plugin:

https://github.com/DeaDBeeF-Player/deadbeef/blob/b4e867ec3a871a266ec18ad926caae58ea4cefd2/plugins/lastfm/lastfm.c

Around line 242-247, there is something about version 1.2, but I'm not sure if that's the protocol or something else.

**Comment by mannp** [comment]
For simple scrobbler did you use the full web URL in both the base URL and api URL on simple scribbler?

Thanks.

**Comment by 1xPdd** [comment]
> For simple scrobbler did you use the full web URL in both the base URL and api URL on simple scribbler?

ListenBrainz URL: `https://domain.tld`
ListenBrainz API URL: `https://domain.tld/apis/listenbrainz`

**Comment by krateng** [comment]
It should work, but you might wanna double-check that IP address.

**Comment by mannp** [comment]
With some modified files Squeezebox / Slimserver users can scrobble to maloja :)

**Comment by rfvgyhn** [comment]
[mpdscribble](https://github.com/MusicPlayerDaemon/mpdscribble) works with the legacy endpoint.

```
[maloja]
url = https://your.domain/apis/audioscrobbler_legacy/
username = u
password = maloja-api-key
journal = /home/user/.mpdscribble/maloja.journal
```

**Comment by FoxxMD** [comment]
I created a (dockerized) javascript app that [scrobbles a user's Spotify listening history to Maloja.](https://github.com/FoxxMD/multi-scrobbler) Since it is directly polling Spotify's API it is independent of listening platform. It uses Maloj's native API.

EDIT: Also supports scrobbling from Plex/Tautulli

**Comment by 1xPdd** [comment]
> [mpdscribble](https://github.com/MusicPlayerDaemon/mpdscribble) works with the legacy endpoint.
> 
> ```
> [maloja]
> url = https://your.domain/apis/audioscrobbler_legacy/
> username = u
> password = maloja-api-key
> journal = /home/user/.mpdscribble/maloja.journal
> ```

This is working great mpd + ncmpcpp! Thanks.

**Comment by moyitpro** [comment]
I made a native scrobbler using the native API for macOS (iTunes/Music, Swinsian)

https://github.com/Atelier-Shiori/OngakuKiroku

**Comment by 1xPdd** [comment]
Don't know how I missed this, but Quod Libet is working very nicely with scrobbling service set to "Other", username as anything, and `/apis/audioscrobbler_legacy` tacked on the end of the URL.

**Comment by isolume** [comment]
any info on browser extensions (firefox, chromium) that work?
 i used to use web scrobbler, but it doesn't seem to work with my maloja server even though it has a dedicated maloja login

**Reply by rhamzeh** [reply]
For [Web Scrobbler](https://github.com/web-scrobbler/web-scrobbler), are you using:
`$MALOJA_URL/apis/mlj_1/newscrobble` as the endpoint? I wasn't, but when I changed it to this, it worked.

I found this after checking out the nifty Maloja API Explorer, you can try it on your installation at: `$MALOJA_URL/api_explorer`

**Reply by isolume** [reply]
oh thanks, ill try that out right now. I assumed it was just /apis/mlj_1/ , as that has worked for other scrobblers

**Reply by isolume** [reply]
update: it still does not seem to be scrobbling after this change
i assume this is more of a web scrobbler issue than a maloja one though, so ill try asking there

**Comment by duckcantcode** [comment]
i am trying to build a maloja plugin for Winamp/WACUP, not sure how it'll go, join my maloja unofficial server [discord.gg/9vS7eNu73r](https://discord.gg/9vS7eNu73r) if you know anything about developing Winamp plugins or DLLs in Windows

**Comment by duckcantcode** [comment]
audiosurf the game is a (partially) working scrobbler client now with my new mod 😎 https://github.com/duckfromdiscord/as-scrobbler https://github.com/duckfromdiscord/audiosurfscrobblerlib

**Comment by subjut** [comment]
[Foobar2000](https://www.foobar2000.org/) on MS Windows will get support for Maloja through the [foo_listenbrianz2](https://github.com/phw/foo_listenbrainz2) plugin using the ListenBrainz API. Make sure to peruse the readme properly as the endpoint URL is a little different.

**Comment by DanGarion** [comment]
I use Pano Scrobbler on my Android device. It is great I have it Scrobble to both Last.fm and my Maloja instance that I run on my server.

**Reply by cairobraga** [reply]
sorry to intrude, but can you tell me how you got Pano Scrobbler to work with your Maloja instance, @DanGarion? I've been unsuccessful in my attempts. thanks in advance!

**Reply by DanGarion** [reply]
You will need to forward your port (or use a reverse proxy method) to expose Maloja if you want to scrobble while outside of your network).
Go into the admin panel of Maloja
Go the API Keys Section
Create your key
The API details are here https://github.com/krateng/maloja/blob/master/API.md
Go into Panoscrobbler,
Go to Configure
Under Services locate Custom Listenbrainz
Use your url such as https://MYWEBSITE.URL//apis/listenbrainz
And then paste in your key
And then the Magic should happen.

**Reply by cairobraga** [reply]
I did all that, and there is no magic happening for me: it gives me an instant refusal saying "service not implemented".

**Reply by DanGarion** [reply]
Can you get to this page on your server? You may need to do http intead of https and also make sure you are using the appropriate port in your url (ie 42010)

https://ENTERYOURIPORINTERNALDOMAINHERE/api_explorer ?

**Reply by cairobraga** [reply]
I can get to this page: https://maloja.cairobraga.com/api_explorer

what should I do after?

**Reply by DanGarion** [reply]
That is good, then you should be able to get Panoscrobbler to work. Your URL, no user name and the API key you create. This should be your URL link https://maloja.cairobraga.com/apis/listenbrainz
I know it says invalid API when you go direct (and when testing with Postman) but it works when the app is used... don't ask me how or why it works. I was pulling hair trying to get postman to work with it for testing.

**Reply by DanGarion** [reply]
Make sure you use Custom Listenbrainz
![image](https://github.com/krateng/maloja/assets/1666375/5b3fde65-0619-4eb2-a62e-893928f56248)
Then enter the URL and the key
![image](https://github.com/krateng/maloja/assets/1666375/b1a913df-0491-4875-82b3-e2bad2fc5387)

**Reply by DanGarion** [reply]
You can also take a look at your logs folder in the Maloja config folder that you have. Take a look at apis.log and see if you have validate-token entry in there. You might see some errors after that but I don't think that matters much.
![image](https://github.com/krateng/maloja/assets/1666375/fe206445-7e48-4e4a-9085-27db2d6b3be0)

If things are working you will see submit-listens when the app submits your listens to Maloja.

**Reply by cairobraga** [reply]
there's no reaction on apis.log and Pano automatically rejects the verification, it doesn't even ping the server, it simply gives me a "forbidden" icon with toast message "service not implemented". I think my phone is the problem. is it a Xiaomi thing, maybe? I see people online making it work and I simply can't.

**Reply by DanGarion** [reply]
Can you browse to your URL from your phone? That is strange. If you want to message me via DM and provide me an API key I can test on one of my devices but only if you feel comfortable with me doing that.

**Reply by DanGarion** [reply]
Did you end up getting it to work?

**Reply by maluvola** [reply]
Hello there. I'm a noob. I managed to use a reverse proxy with loophole and zrok. But with them I couldn't make a custom and static domain like you, like maloja.mydomain.something. Loophole generated a myname.loophole.site, but it isn't reserved, it can be claimed by anyone if I close it. Zrok can generate a reserved static token for some time (and I failed in doing it and gave up, but it isn't custom anyways, it's random). How did you do that? What do you recommend for that purpose? Thanks.

**Reply by DanGarion** [reply]
Since I'm using Docker I highly recommend using SWAG it takes some time to figure it out but it has worked for my setup and everything I do. https://docs.linuxserver.io/general/swag/
This might also help with you to understand how to get it working... https://docs.ibracorp.io/swag-2/

**Comment by duckcantcode** [comment]
I will add that iOS is pretty much unable to use something like Pano Scrobbler since apps cannot access the currently playing song (trust me, I've tried). The only native Maloja solution on the App Store is Web Scrobbler which only works for the browser as an extension. As far as I know there are no other apps that support Maloja on the Store. A Subsonic client like [iSub](https://github.com/einsteinx2/iSubMusicStreamer) and a Listenbrainz-compatible Subsonic server are pretty much the only solution I can think of on iOS.

**Comment by thagoat** [comment]
I'm using Navidrome in docker and Symfonium on Android. Works perfectly. In my Navidrome docker-compose I added environment values:
ND_LISTENBRAINZ_ENABLED: "true"
ND_LISTENBRAINZ_BASEURL: "http://<maloja_server.url:42010/apis/listenbrainz/1/"
ND_LISTENBRAINZ_APIKEY: "my-api-key-from-maloja"

Note the /1/ in the BASEURL. That's key to getting Navidrome to scrobble correctly.

**Reply by DanGarion** [reply]
The Panoscrobble app doesn't require /1/ in the URL at least it doesn't in my config...

**Comment by morini3andahalf** [comment]
I've used last fm and libre fm for a good few years but looking to try and break cloud dependency where possible. Started playing with maloja yesterday and it seems to work great (running locally, docker-compose). Currently using simple scrobbler on my android phone and it scrobbles to all three platforms quite happily.

I was wondering, is there any way to get a "now playing" working like libre FM? Better still if able to embed that info into another web page.

Great project, kudos to the author.

**Comment by duckcantcode** [comment]
Working on compiling an exhaustive list of scrobblers with native maloja support at https://github.com/duckfromdiscord/awesome-maloja. Let me know if there are any I have missed.
